src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesCcm.Android.cs

```diff
@@ -0,0 +1,178 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.Win32.SafeHandles;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class AesCcm
+    {
+        private byte[] _key;
+
+        [MemberNotNull(nameof(_key))]
+        private void ImportKey(ReadOnlySpan<byte> key)
+        {
+            // OpenSSL does not allow setting nonce length after setting the key
+            // we need to store it as bytes instead
+            _key = key.ToArray();
+        }
+
+        private void EncryptInternal(
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> plaintext,
+            Span<byte> ciphertext,
+            Span<byte> tag,
+            ReadOnlySpan<byte> associatedData = default)
+        {
+            // Convert key length to bits.
+            using (SafeEvpCipherCtxHandle ctx = Interop.Crypto.EvpCipherCreatePartial(GetCipher(_key.Length * 8)))
+            {
+                Interop.Crypto.CheckValidOpenSslHandle(ctx);
+
+                if (!Interop.Crypto.EvpCipherSetTagLength(ctx, tag.Length))
+                {
+                    throw Interop.Crypto.CreateOpenSslCryptographicException();
+                }
+
+                Interop.Crypto.EvpCipherSetCcmNonceLength(ctx, nonce.Length);
+                Interop.Crypto.EvpCipherSetKeyAndIV(ctx, _key, nonce, Interop.Crypto.EvpCipherDirection.Encrypt);
+
+                if (associatedData.Length != 0)
+                {
+                    if (!Interop.Crypto.EvpCipherUpdateAAD(ctx, associatedData))
+                    {
+                        throw Interop.Crypto.CreateOpenSslCryptographicException();
+                    }
+                }
+
+                byte[]? rented = null;
+                try
+                {
+                    Span<byte> ciphertextAndTag = stackalloc byte[0];
+                    // Arbitrary limit.
+                    const int StackAllocMax = 128;
+                    if (ciphertext.Length + tag.Length <= StackAllocMax)
```

Comment:
This looks like it could overflow to a negative value, which will be less than `StackAllocMax`. In that case we'll end up doing `stackalloc byte[-someNumber]`. The CLR treats the input to `stackalloc` as unsigned, so it'll result in a huge stackalloc and crash the process.

This should either be guarded or at the very least `checked`.